core/unix/os.c

```diff
@@ -1087,6 +1087,15 @@ get_application_name_helper(bool ignore_cache, bool full_path)
     return (full_path ? executable_path : executable_basename);
 }
 
+#ifdef CLIENT_INTERFACE
+void
+set_client_error_code(dr_error_code_t error_code)
```

Comment:
Should this take in the dcontext, since many routines already have it, and allow NULL or GLOBAL_DCONTEXT (berst to check for both to make it easy for callers to blindly pass their local var dcontext) in which case it calls get_thread_private_dcontext()?

Reply:
Hmm, so for the cases considered in this PR, actually, drcontext is not readily available. Getting the drcontext via this function reduces code. 

Also, most of the time, drcontext is obtained only upon error. Therefore harnessing the getting of drcontext out of the branch where set_client_error_code() would be called would result in a slight perf hit.

Reply:
For these, they don't have dcontext, true: though I think that most other routines do?

> Therefore harnessing the getting of drcontext out of the branch where set_client_error_code() would be called results in a slight perf hit.

Not sure I follow: the caller of set_client_error_code() only passes dcontext if they have it readily available in a local; else they pass NULL and this function looks it up for them.

Seems fine to leave it as it is w/o taking in the dcontext: since internal we can always change it later if we end up with a whole bunch of calls who have the dcontext.

Reply:
Ah, I see. So you mean enable the passing of drcontext as a param but if it is set to NULL, have set_client_error_code() call get_thread_private_dcontext().

Reply:
Done.

Reply:
> Ah, I see. So you mean enable the passing of drcontext as a param but if it is set to NULL, have set_client_error_code() call get_thread_private_dcontext()?

Yes, and to also do so for GLOBAL_DCONTEXT so callers don't have to.